### ISSUE TITLE
Add support for PHP 5.5 on Ubuntu 14.04

### ISSUE THREAD

**Issue by TomKeur** [DataDog/dd-trace-php#625]
Since we're running some legacy systems on Ubuntu 14.04 which ships with PHP 5.5, we want to get support for PHP 5.5.

I know PHP 5.5 is officially End Of Life, but since we're running Ubuntu 14.04 with [Ubuntu Extended Security Maintenance](https://ubuntu.com/esm), we're still getting security updates for PHP 5.5.

I know when we will upgrade to PHP 5.6 the extension will work, but upgrading to PHP 5.6 has no effort for us. Because when we do, we're losing our security patches since PHP 5.6 is End Of Life too.

My issue is related to #616.

PS: We're upgrading to PHP 7.x, but this is a steep road up.

**Comment by labbati** [comment]
Thanks @TomKeur for opening this issue. You made a very good point and we definitely want to support your use case.
We plan to start working on this during this quarter. While I cannot commit to deliver by EOQ, we will at least be able to provide a better estimate once with start dig into this.

As a side note a good news: we (@morrisonlevi) was able to build for 5.5 with no problems and a decent amount of tests already pass. This might let us being positive.

**Comment by TomKeur** [comment]
Thanks for your reply @labbati.
This is great news :)

**Comment by morrisonlevi** [comment]
What is your webserver setup? Is this Apache, nginx..? When this work gets scheduled it would be nice to have this information.

**Comment by TomKeur** [comment]
We're using Apache as mod_php, classic (since it's PHP 5.5 :)).

**Comment by TomKeur** [comment]
@morrisonlevi Hi, any news on this? Looking really forward to start using this in production.
Thanks!

**Comment by TomKeur** [comment]
@labbati @morrisonlevi Can I please have an update? Since we're really in need of this.

**Comment by labbati** [comment]
Hi @TomKeur, I am bringing in this with our PM and get back to you, but let me give you some context about the work that is required to be done.

The PHP tracer hooks into the opcode handlers for function/method calls. Then from within our hooks we forward the call to the original method. This has, for us, the huge disadvantages that 1) there might potentially be stability related issues when we work along side other extensions and 2) that we have to do a lot of heavy lifting on behalf of the VM, and this can be crazy different for different versions of PHP.

In order to support a new version of PHP, we need first to change the way we use such hooks. The work is beginning this sprint (or early next sprint). Once that part is done, I can back to you with a more certain ETA as, from that point on, will only be a matter to setup CI and containers to properly test 5.5.

@morrisonlevi anything to add?

**Comment by TomKeur** [comment]
Thanks for the explanation, this explains why it's not that easy! I thought it would be easier :), since PHP 5.4 and PHP 5.6 are already supported.

**Comment by TomKeur** [comment]
@labbati @morrisonlevi Can I please have an update? Unfortunately, this is feature is still necessary for us.

**Comment by morrisonlevi** [comment]
We are still working on changing the way we hook into the VM opcodes. The approach seems to be working well on PHP 7, but it is less clear on PHP 5. In other words, I can't provide an ETA. We discussed this in a recent meeting and Ubuntu 14.04 ESM goes until April 2020 so we definitely want to support it; it's just man hours and prioritization from here.

To help speed up the process for when we get to it: what frameworks or major libraries do you use? This helps us know which integrations to prioritize and test with.

**Comment by TomKeur** [comment]
> To help speed up the process for when we get to it: what frameworks or major libraries do you use? This helps us know which integrations to prioritize and test with.

We're only using Symfony 2.

**Comment by SammyK** [comment]
Hey @TomKeur! I just wanted to give an update on this. Although we still don't have an exact timeline yet, we've made good progress toward supporting PHP 5.5 and a ballpark estimate for PHP 5.5 support will be around the end of June. We'll keep you updated! :)

**Comment by kipit** [comment]
Hello Datadog team! Any update on this support?

**Comment by morrisonlevi** [comment]
Off the top of my head I believe it's scheduled to be worked on within a month. @labbati, can you confirm?

Note that some of the prerequisites have been and are still being worked on. For example, version 0.48.0 will change how we trace function calls on PHP 5 and that's the last major prereq for PHP 5.5 that I am aware of.

**Comment by labbati** [comment]
Work is tentatively scheduled for weeks Aug 17 and 24. So far we are on
track.
As Levi mentioned this is blocked by 0.48.0 which include all the
prerequisites that were blocking this task.

**Comment by morrisonlevi** [comment]
PHP 5.5 support has been merged in. It still needs more tests before it's officially supported, but wanted all of you to know it's coming soon, probably in v 0.49.0.

**Comment by SammyK** [comment]
PHP 5.5 is now supported in [0.49.0](https://github.com/DataDog/dd-trace-php/releases/tag/0.49.0). Thank you to everyone for your patience.
